File: nexus-bank-ui/src/services/authService.ts
import api from './api';
import type { AuthResponse } from '../types';

export const authService = {
  async login(request: any): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/login', request);
    return response.data;
  },

  async register(request: any): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/register', request);
    return response.data;
  },

  async logout(): Promise<void> {
    try {
      await api.post('/auth/logout');
    } catch {
      // Server-side revocation is best-effort; always clear local state
    } finally {
      localStorage.removeItem('token');
      localStorage.removeItem('user');
    }
  }
};

File: src/main/java/com/nexus/core/auth/AuthService.java
package com.nexus.core.auth;

import com.nexus.core.auth.dto.AuthResponse;
import com.nexus.core.auth.dto.LoginRequest;
import com.nexus.core.auth.dto.RegisterRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}
